src/main/java/org/checkstyle/autofix/recipe/Header.java

```diff
@@ -69,60 +65,53 @@ public String getDescription() {
 
     @Override
     public TreeVisitor<?, ExecutionContext> getVisitor() {
-        final String licenseHeader = extractLicenseHeader(config, charset);
+        final String licenseHeader = extractLicenseHeader(config);
         final List<Integer> ignoreLines = extractIgnoreLines(config);
         return new HeaderVisitor(violations, licenseHeader, ignoreLines);
     }
 
-    private static String extractLicenseHeader(Configuration config, Charset charset) {
+    private static String extractLicenseHeader(CheckConfiguration config) {
         final String header;
         try {
-            if (hasProperty(config, HEADER_PROPERTY)) {
+            if (config.hasProperty(HEADER_PROPERTY)) {
                 header = config.getProperty(HEADER_PROPERTY);
             }
             else {
                 final Charset charsetToUse;
-                if (hasProperty(config, CHARSET_PROPERTY)) {
+                if (config.hasProperty(CHARSET_PROPERTY)) {
                     charsetToUse = Charset.forName(config.getProperty(CHARSET_PROPERTY));
                 }
+                else if (config.getParent().hasProperty(CHARSET_PROPERTY)) {
+                    charsetToUse = Charset.forName(config
+                            .getParent().getProperty(CHARSET_PROPERTY));
+                }
                 else {
-                    charsetToUse = charset;
+                    charsetToUse = Charset.defaultCharset();
```

Comment:
Does it make sense to add `getPropertyOrDefault` into config class to handle such cases?